### PR TITLE
Allow passing custom mig-parted configmap data in helm chart 

### PR DESCRIPTION
**Pull request by frittentheke**
Closes: #802

### CODE REVIEW

**Review by shivamerla — APPROVED** [review]
lgtm

File: deployments/gpu-operator/values.yaml [hunk 1]
```diff
@@ -358,9 +358,42 @@ migManager:
...
+  #         mig-enabled: true
+  #         mig-devices:
+  #           "3g.40gb": 1
+  #           "4g.40gb": 1
   config:
-    name: "default-mig-parted-config"
     default: "all-disabled"
+    # Create a ConfigMap (default: false)
+    create: false
+    # ConfigMap name (either exiting or to create a new one with create=true above)
```
Comment (cdesiniotis) [inline deployments/gpu-operator/values.yaml]: nit: typo
```suggestion
    # ConfigMap name (either existing or to create a new one with create=true above)
```

File: deployments/gpu-operator/values.yaml [hunk 2]
```diff
@@ -358,9 +358,42 @@ migManager:
...
+    name: "default-mig-parted-config"
```
Comment (cdesiniotis) [inline deployments/gpu-operator/values.yaml:394]: Let's make the default name empty to align with the values for the device-plugin. 
```suggestion
    name: ""
```

File: deployments/gpu-operator/templates/mig_config.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{{- if and (.Values.migManager.config.create) (not (empty .Values.migManager.config.data)) }}
```
Comment (tariq1890) [inline deployments/gpu-operator/templates/mig_config.yaml:1]: Won't this suffice?
```suggestion
{{- if and (.Values.migManager.config.create) ( .Values.migManager.config.data) }}
```

**Comment by cdesiniotis** [comment]
@frittentheke can you amend the commit message to better indicate the change introduced? My suggestion: `Allow passing custom mig-parted configmap data in helm chart`.

File: deployments/gpu-operator/values.yaml (hunk 2 above)
Comment (tariq1890) [reply deployments/gpu-operator/values.yaml:394]: @cdesiniotis I am unsure if we should do this, as we deploy this configmap by default when the mig-manager is enabled.

Comment (tariq1890) [reply deployments/gpu-operator/values.yaml:394]: Ah ok, I do see [here](https://github.com/NVIDIA/gpu-operator/blob/7068fec931e78af2994f82a6ecfea55cbf737b48/controllers/object_controls.go#L1744) that we inject `default-mig-parted-config` when it's unspecified

Please disregard my comment

File: deployments/gpu-operator/values.yaml (hunk 1 above)
Comment (frittentheke) [reply deployments/gpu-operator/values.yaml]: I copied this typo over from the existing comment :see_no_evil: 
I shall fix them both.

**Comment by frittentheke** [comment]
@shivamerla @cdesiniotis @tariq1890 

Thanks a bunch for your review and feedback!
I just pushed another revision fixing your remarks - PTAL.

**Review by tariq1890 — APPROVED** [review]

**Comment by frittentheke** [comment]
@tariq1890 @shivamerla there is are two e2e tests failing due to a missing ssh_key:
> ssh key not provided

Could you kindly have a look into this? Would love to have this PR succeed in tests to be ready to merge.

File: deployments/gpu-operator/values.yaml (hunk 2 above)
Comment (cdesiniotis) [reply deployments/gpu-operator/values.yaml:394]: @frittentheke I do not see this comment being addressed. Can you update this as I have suggested? The `name` field here should only be used when configuring a custom ConfigMap, so defaulting to `default-mig-parted-config` does not make sense.

**Review by cdesiniotis — CHANGES_REQUESTED** [review]
@frittentheke requesting one more change before we can merge this.

**Comment by cdesiniotis** [comment]
> @tariq1890 @shivamerla there is are two e2e tests failing due to a missing ssh_key:
> 
> > ssh key not provided
> 
> Could you kindly have a look into this? Would love to have this PR succeed in tests to be ready to merge.

@frittentheke this is expected as our e2e tests require that your Github repo has valid AWS credentials configured. We can merge this PR without these tests passing.

**Comment by klueska** [comment]
I haven't looked closely, but does this mimic what we do for the device plugin configs?

**Comment by cdesiniotis** [comment]
> I haven't looked closely, but does this mimic what we do for the device plugin configs?

Yes, this mimics what we have for the device-plugin.

**Comment by frittentheke** [comment]
> @frittentheke requesting one more change before we can merge this.


@cdesiniotis This part is actually unchanged to how it was / is prior to this PR and I thought this was fine as it is the same name used by the default config map - see comment by  https://github.com/NVIDIA/gpu-operator/pull/803/files/3e655d7704070c2cc9d7e8afad4bbf08bb774bf0#r1676570719

I gladly change it to some other name if you want me to.

**Comment by cdesiniotis** [comment]
> @cdesiniotis This part is actually unchanged to how it was / is prior to this PR 

I overlooked this in a prior review. Thanks for pointing this out. I can update the name of this field in a follow-up. Thanks for the contribution!

**Comment by cdesiniotis** [comment]
I have opened https://github.com/NVIDIA/gpu-operator/pull/849 to align the `config.name` and `config.data` fields with the device-plugin config.

**Pull request merged**